ddtrace/contrib/httplib/patch.py

```diff
@@ -67,6 +70,17 @@ def _wrap_putrequest(func, instance, args, kwargs):
         span.set_tag(ext_http.METHOD, method)
     except Exception:
         log.debug('error applying request tags', exc_info=True)
+    return func(*args, **kwargs)
+
+
+def _wrap_putheader(func, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin:
+        return func(*args, **kwargs)
+
+    span = getattr(instance, '_datadog_span', None)
```

Comment:
@Kyle-Verhoog we should be able to do the following right?

```python
ctx = pin.tracer.get_call_context()
span = ctx.get_current_span()
```

If so, is one way better than the other?

Reply:
Oops! Didn't see this. (Github needs a better mention notification system).

It seems like here @labbati is manually handling the span retrieval, this could be due to the internals httplib. In this case the span won't be in the context.

Reply:
@Kyle-Verhoog Actually the span is not `finish`-ed when this method is called, so looking at the code what @brettlangdon suggested should work. I will take the opportunity to spend some time looking into the under-the-hood of how the context actually work as I miss a lot of knowledge there. Thanks both.